core/embed/rust/src/ui/display/font.rs

```diff
@@ -22,40 +29,30 @@ pub struct Glyph {
     pub bearing_x: i16,
     /// Top-side vertical bearing
     pub bearing_y: i16,
-    data: &'static [u8],
+    data: &'a [u8],
 }
 
-impl Glyph {
-    /// Construct a `Glyph` from a raw pointer.
+impl<'a> Glyph<'a> {
+    /// Creates a new `Glyph` from a byte slice containing font data.
     ///
-    /// # Safety
-    ///
-    /// This function is unsafe because the caller has to guarantee that `data`
-    /// is pointing to a memory containing a valid glyph data, that is:
-    /// - contains valid glyph metadata
-    /// - data has appropriate size
-    /// - data must have static lifetime
-    pub unsafe fn load(data: *const u8) -> Self {
-        unsafe {
-            let width = *data.offset(0) as i16;
-            let height = *data.offset(1) as i16;
-
-            let data_bytes = match constant::FONT_BPP {
```

Comment:
you should use `calculate_glyph_size` here and `assert!` that the data length matches (or maybe debug_assert? debug_assert is fine for builtin fonts, unsure about the external ones)

Reply:
74a84516f9303f0b95cdd9a55e5457fb06898592